Fixed unused kwargs in reflex.App

Snaipergelka opened the pull request:
### Type of change
- [x] Bug fix (non-breaking change which fixes an issue)

Fixed kwargs `sio` and `socket_app` are not used during reflex.App initialization

Closes #3081

masenf reviewed (changes requested):
Thanks for the contribution and tracking down that mysterious bug!

I have a couple of suggestions to improve the patch. The team has been wondering if exposing the `sio` externally is a good idea (yes it was documented, via generated docs), or if we should just expose the parameters as configuration knobs, like cors_allowed_origins.

For now, I'm leaning toward allowing the user to supply their own AsyncServer object.

Snaipergelka commented on reflex/app.py:
```diff
@@ -223,21 +223,24 @@ def setup_state(self) -> None:
+                max_http_buffer_size=constants.POLLING_MAX_HTTP_BUFFER_SIZE,
+                ping_interval=constants.Ping.INTERVAL,
+                ping_timeout=constants.Ping.TIMEOUT,
+            )
```
Thank you for feedback. I updated patch.

Snaipergelka commented:
Updated patch in [commit](https://github.com/reflex-dev/reflex/pull/3170/commits/fbb527056591284174dc56ca442d6b2aaed5a5d3). Made `socket_app` local variable (deleted from the `reflex.App` args) and added check for `async_mode=asgi` for custom `sio` in `reflex.App`.
